Cybersecurity_Tools/DEEPFAKE/extension/backend/main.py

```
@@ -0,0 +1,305 @@
+        confidence = float(max(avg_pred))
+        label = "Fake" if avg_pred[0] > avg_pred[1] else "Real"
+        
+        return {
+            "result": label,
+            "confidence": confidence,
+            "frames_analyzed": len(predictions),
+            "error": None
+        }
+    except Exception as e:
+        logger.error(f"Video detection error: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Video processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)